plugins/widget/plugin.js

```diff
@@ -2053,9 +2053,62 @@
 	//
 	// @param {CKEDITOR.htmlParser.element} el
 	function cleanUpWidgetElement( el ) {
+		// Preserve initial and trailing space by replacing white space with &nbsp; (#605).
+		var firstTextNode = getFirstTextNode( el ),
+			lastTextNode = getLastTextNode( el );
+
+		if ( firstTextNode || lastTextNode && el.attributes[ 'data-cke-widget-data' ] ) {
+			// Add attribute to element if any white space will be replaced.
+			if ( firstTextNode.value.match( /^\s/g ) || lastTextNode.value.match( /\s$/g ) ) {
+				el.attributes[ 'data-cke-widget-white-space' ] = 1;
+			}
+
+			// Check whether the value of the first text node contains white space at the beginning and replace it with &nbsp;.
+			if ( firstTextNode.value && firstTextNode.value.match( /^\s/g ) ) {
+				firstTextNode.parent.attributes[ 'data-cke-white-space-first' ] = 1;
+				firstTextNode.value = firstTextNode.value.replace( /^\s/g, '&nbsp;' );
+			}
+
+			// Check whether the value of the last text node contains white space at the end and replace it with &nbsp;.
+			if ( lastTextNode.value && lastTextNode.value.match( /\s$/g ) ) {
+				lastTextNode.parent.attributes[ 'data-cke-white-space-last' ] = 1;
+				lastTextNode.value = lastTextNode.value.replace( /\s$/g, '&nbsp;' );
+			}
+		}
+
 		var parent = el.parent;
-		if ( parent.type == CKEDITOR.NODE_ELEMENT && parent.attributes[ 'data-cke-widget-wrapper' ] )
+
+		if ( parent.type == CKEDITOR.NODE_ELEMENT && parent.attributes[ 'data-cke-widget-wrapper' ] ) {
 			parent.replaceWith( el );
+		}
+	}
+
+	// Get first text node (#605).
```

Comment:
I think ticket number is not needed here as this function is used by the code which already contains proper description. Same for the second function below.